### DISCUSSION TITLE
Elrs tm playback with simulator:  (sbus/ltm ?)tm shown instead

### DISCUSSION THREAD

**Discussion by alccrl** [EdgeTX/edgetx#3898] [Q&A]
Hi,

configuring (just compiled) edgetx companion and simulator master branch code (2.10 version shown), with an scard copied from real transmitter x9lite, with real flight logs, I get incorrect tm sensor names, not elrs names.
Example: I get RSSI, I should get RSS1 and RSS2. It gives me playing pack log low tx battery (2.2 v), should be 7.x v. I use arducopter sw in FC.
On real transmitter I get back correct names and values while flying.

Do I miss some configuration step in companion/simulator ?

Thanks in advance for help.
I can write more details if necessary.

Alfredo

**Comment by pfeerick** [comment]
Simulator only knows about mainly Frsky telemetry - there is no support for ELRS telemetry... yet.

**Reply by alccrl** [reply]
Ok ....thanks,

is there anyway to couple, if You know, arducopter with elrs receiver and x9lite tx in order to have "everything" working?
Most important is lua script for screen tm, it does not work with elrs+lua(arducopter).

Need I go back to Opentx ?...

Alfredo

**Reply by pfeerick** [reply]
What exactly are you actually trying to do? Simply viewing the logs (File -> View Log File...) which should work, or replay them in simulator (which I expect only have some fields working, the sensors that happen to have the same name). 

I don't use the X9Lite or arducopter, so can only give some generic advice. What Lua are you using that is not working? Is the ELRS Lua itself working? If you are trying to use Yappu telemetry, make sure you are using the latest version, as 2.00 seems to be the recommended version. Is there any particular reason you are using ETX 2.10 (unstable), rather than 2.9-RC1?

**Reply by alccrl** [reply]
Yappu lua seems to have problems too, with elrs (as declared on site).

I did too many changes at the same time ...
both hw (from sbus to elrs rx+tx modules) and rc (from opentx to edgetx). 
Moreover I switched from inav to arducopter  on fc, but all of these  except lua screen work, this is why I got stuck in simulator playback and asked queation about it.
I am using edgetx 2.10 because to be honest I just got last version to test build env (linux ubuntu 23.04), I am not aware of which versions are stable or not.
I found a few difficulties on 23.04 because .sh  build  instruction file is tuned for ubuntu  20.04, quite obsolete, anyway I succeded after modifying the libclang search script file and after search for equivalent packages from 20.04 to 23.04.

Elrs is so much "advertised" , that I did not realize that so many things are yet not ready for it.

Alfredo

**Comment by alccrl** [comment]
@pfeerick :
"Is there any particular reason you are using ETX 2.10 (unstable), rather than 2.9-RC1?"

I just tried to download and use simulator 2.9.rc2 binaries, but I did not find it in assets (linux version contain appimage only for companion, not simulator).
After googling for AppImage system I found a way to extract simulator binary from the Appimage file, but it fails with a qt missing symbol.

Then ... 
I downloaded source 2.9rc2 and tried to compile, but thirdparties were missing.
Then ...
I searched for 2.9 rc2 git TAG, but I found only zipped code with no thirdparty code and git clone dialog has no correct version referred (it refers master branch, I guess), so no way to download exact (complete) 2.9rc2 bundle.

In other words ...,
there is some quirk setup in git/github, in zip 2.9rc2 code thirdparties are not included correctly.
I use sometimes git,but when I checkout a TAG, generally all the compilation stuff is included.
For edgetx it seems to work in a different way...

So, 
shall I by hand/diff/compare see differences between nigthly build and 2.9rc2 sources to create a "compilable/linkable" code 2.9rc2 ?
I don't think this to be very good practise....
some other way must exist, otherwise how can exist binary simulator assets ?

Any advice ?

Alfredo

**Comment by alccrl** [comment]
Found a way ...:

git clone --recursive -b 2.9rc2 

Tm window shows sbus data, but in simulator source code are present also frsky tm strings (RSS1, RSS2 and so on). I don't think the problem to be in elrs, for some reason tm playback dialog shows  sbus tm, csv log file shows crsf tm.
There is a mismatch.

I guess this is a bug or incomplete implementation, not a missing feature.

Alfredo
